### DISCUSSION TITLE
Padding size solution for small images

### DISCUSSION THREAD

**Discussion by sardnar** [Project-MONAI/MONAI#6518] [Q&A] [answered]
I have a problem that I don't have a solution to fix as my images are already resized to specific dimension. I am using HoverNet Model and my images has been cropped to special dimensions. Now, running inference, I am getting.

```
RuntimeError: Padding size should be less than the corresponding input dimension, but got: padding (12, 12) at dimension 3 of input [1, 1, 5333, 5]
``

What are the possible workarounds for that one?

Thank you

**Comment by KumoLiu** [comment]
Hi @sardnar, could you please share more information such as how you cropped your data, then I could take a deep look at it, thanks!

**Reply by sardnar** [reply]
I just cropped them 5000x5000 without padding unless it is at one of the edges, it be as small as it could. @KumoLiu is that enough?

**Reply by KumoLiu** [reply]
Could you please share what transform you get the error, and also the whole error message? Maybe also your transform chain. It's not easy to find the reason from only one line message. :)
Thanks!

**Reply by sardnar** [reply]
This is coming from HoVerNetInstanceMapPostProcessingd() during inference.  In this particular case he image has height of 5 pixels

```
applying transform <monai.transforms.compose.Compose object at 0x2ab2d17da1d0>
RuntimeError: Padding size should be less than the corresponding input dimension, but got: padding (12, 12) at dimension 3 of input [1, 1, 5333, 5]

The above exception was the direct cause of the following exception:

RuntimeError: applying transform <monai.apps.pathology.transforms.post.dictionary.HoVerNetInstanceMapPostProcessingd object at 0x2ab2d17d9600>
```

I am just not sure where the 12 padding is coming from?

If you can guide me @KumoLiu would be awesome.

**Reply by KumoLiu** [reply, marked as answer]
Hi @sardnar, thanks now I can reproduce the error.
I suppose you were just following the default value set in the tutorial.
https://github.com/Project-MONAI/tutorials/blob/3dfe68f941d2768c723fdbfbfb54bc41f265cf0c/pathology/hovernet/inference.py#L83
The default value for the `sobel_kernel_size` is 21, and the padding size is calculated from it. So If your input size only has 5, I would suggest you set `sobel_kernel_size` smaller.
https://github.com/Project-MONAI/MONAI/blob/e084885c6d498ab02df8107d23aa159f742a2cdd/monai/networks/layers/simplelayers.py#L246

Hope it can help you, thanks!
